src/containers/SignupForm/SignupForm.js

```diff
@@ -31,36 +33,50 @@ class Signup extends Component {
       [name]: value
     })
   }
+  componentDidMount() {
+    const searchParams = new URLSearchParams(this.props.location.search)
+    this.state.code = searchParams.get('code') ? searchParams.get('code') : ''
 
+  }
   handleSubmit = async e => {
     e.preventDefault()
-    const { username, email, password } = this.state
-
+    const { username, email, password, code } = this.state
+    console.log(code)
+    if(code == ''){
+      this.setState({ errors: [{ message: '目前只可通过邀请链接注册'}] })
+      return 
+    }
     try {
       const { errors } = await this.props.signupMutation({
         variables: {
           username,
           email,
-          password
+          password,
+          code
         },
         update: (cache, { data }) => {
           const currentUser = data.signup
-
-          // Update currentUser in cache
-          cache.writeQuery({
-            query: GET_CURRENT_USER,
-            data: {
-              currentUser
-            }
+          this.setState({
+            successes: [{ message: '激活邮件已发送,请前往邮箱查看' }]
           })
+          this.state.username = ''
+          this.state.password = ''
+          this.state.email = ''
+          this.state.code = ''
+          // Update currentUser in cache
```

Comment:
Remove unused code.

Reply:
setState these states to disable resubmit

Reply:
I meant the commented code block starting line 66 can be removed.